web/client/src/library/components/plan/PlanApplyStageTracker.tsx

```diff
@@ -90,41 +91,46 @@ export default function PlanApplyStageTracker(): JSX.Element {
           </small>
         </div>
       )}
-      {isFalseOrNil(hasChanges) || isNil(changes) ? (
+      {planAction.isRunning || (isNotNil(changes) && isTrue(hasChanges)) ? (
```

Comment:
I know this isn't exactly new, but what's the difference been changes and hasChanges?

Reply:
it is possible these two don't much, need to refactor it so `hasChanges` is just derived from `changes `